### ISSUE TITLE
[bug] DeepSeek V4 Pro FP8-MX pretraining gives gradient norm 'nan' on GB300

### ISSUE THREAD

**Issue by rhmukundan** [NVIDIA-NeMo/Megatron-Bridge#5306]
### Problem

DeepSeek V4 Pro FP8-MX pretraining is showing **grad norm: nan** across multiple iterations. 
```
grad norm: nan
number of skipped iterations: 0
number of nan iterations: 0

```

The workload otherwise continues running at approximately 21–22 seconds per iteration and 955–992 model TFLOP/s/GPU.
This reproduces using Megatron-Bridge main with Megatron-LM dev.

### Minimal repro

```shell
Check out Megatron-Bridge commit:
573e088c9c6740082c39744e03dc5b009e730ed4

Check out the Megatron-LM submodule commit:
9304a25642eea46d959207934cd62b71194e9f95

Launch the DSV4 Pro workload using:
scripts/performance/setup_experiment.py
```

### Expected behavior

The gradient norm should remain finite during training. If non-finite gradients are encountered, the iteration should be detected and handled consistently, such as being counted as a NaN iteration or skipped according to the configured policy.

### Affected area

area:recipe

### Regression?

Yes

### Logs

```shell
iteration 5/50 | elapsed time per iteration: 21392.5 ms |
learning rate: 1.350000E-04 | global batch size: 4096 |
loss scale: 1.0 | grad norm: nan |
number of skipped iterations: 0 | number of nan iterations: 0

iteration 6/50 | elapsed time per iteration: 21313.4 ms |
learning rate: 1.620000E-04 | global batch size: 4096 |
loss scale: 1.0 | grad norm: nan |
number of skipped iterations: 0 | number of nan iterations: 0

iteration 7/50 | elapsed time per iteration: 21452.0 ms |
learning rate: 1.890000E-04 | global batch size: 4096 |
loss scale: 1.0 | grad norm: nan |
number of skipped iterations: 0 | number of nan iterations: 0

iteration 8/50 | elapsed time per iteration: 22127.0 ms |
learning rate: 2.160000E-04 | global batch size: 4096 |
loss scale: 1.0 | grad norm: nan |
number of skipped iterations: 0 | number of nan iterations: 0
```

**Comment by yaoyu-33** [comment]
@weijiac0619  can you share our working commit combo?

**Comment by huvunvidia** [comment]
Hi @weijiac0619 ,
Just a quick check-in if you can share the working commit combo to resolve the issue.
Thank you.
